Added Xcode 12.5.1 to tests and also made this version the default one.

apps4everyone opened the pull request:
Added Xcode 12.5.1 to tests and also made this version the default one.

danyf90 reviewed (changes requested):
The macos latest GitHub executor does not contain Xcode 12.5.1.
On the other and, the macos.11 executor contains 12.5.1 but doesn't contain 12.2

apps4everyone commented:
> The [macos latest (alias for macos 10.15)](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md) GitHub executor does not contain Xcode 12.5.1.
> On the other and, the [macos 11](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode) executor contains 12.5.1 but doesn't contain 12.2

thx, for the info. Removed 12.2 and added 13. Only to see if tests would be ok.

danyf90 commented:
@apps4everyone you still need to change the `runs-on: macOS-latest` to `runs-on: macOS-11` to use the newer image. 
I know, it's called latest but it's not really the latest :D

danyf90 commented on .github/workflows/tuist.yml:
```diff
@@ -60,10 +60,10 @@ jobs:
     strategy:
       matrix:
-        xcode: ['12.2', '12.4'] # keep 12.2 to ensure backwards compatibility
+        xcode: ['12.4', '12.5.1', '13.0'] # keep 12.2 to ensure backwards compatibility
```
```suggestion
        xcode: ['12.4', '12.5.1', '13.0'] # keep 12.4 to ensure backwards compatibility
```

danyf90 commented on .github/workflows/tuist.yml:
```diff
@@ -60,10 +60,10 @@ jobs:
     strategy:
       matrix:
-        xcode: ['12.2', '12.4'] # keep 12.2 to ensure backwards compatibility
+        xcode: ['12.4', '12.5.1', '13.0'] # keep 12.2 to ensure backwards compatibility
```
@fortmarek @pepibumur @kwridan Do we need to keep 12.2 compatibility? if yes, we should setup another workflow to run on the 10.15 executor

fortmarek replied:
Considering 13.0 is right around the corner, I think it should be fine to sunset 12.2

danyf90 commented on CHANGELOG.md:
```diff
@@ -11,6 +11,11 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 - Allow to pass Cloud authentication token via TUIST_CLOUD_TOKEN even when not CI [#3380](https://github.com/tuist/tuist/pull/3380) by [@danyf90](https://github.com/danyf90)
 
+
+### Changed
+
+- **Breaking** Minimum supported Xcode version bumped to 12.4.
```
Not related to this PR, but could you please merge the 2 `Fixed` section? 🙏

apps4everyone replied:
done

fortmarek commented:
I'd like to fix the CI issue, so closing this and superseded by https://github.com/tuist/tuist/pull/3427.

@apps4everyone I added you as a contributor, so you don't have to create a separate fork next time you do a PR.
